pkg/test/utils/kubernetes.go

```diff
@@ -584,36 +590,48 @@ func WithNamespace(obj runtime.Object, namespace string) runtime.Object {
 }
 
 // WithSpec applies the provided spec to the Kubernetes object.
-func WithSpec(obj runtime.Object, spec map[string]interface{}) runtime.Object {
-	return WithKeyValue(obj, "spec", spec)
+func WithSpec(t *testing.T, obj runtime.Object, spec map[string]interface{}) runtime.Object {
+	res, err := WithKeyValue(obj, "spec", spec)
+	if err != nil {
+		t.Fatalf("failed to apply spec %v to object %v: %v", spec, obj, err)
+	}
+	return res
 }
 
 // WithStatus applies the provided status to the Kubernetes object.
-func WithStatus(obj runtime.Object, status map[string]interface{}) runtime.Object {
-	return WithKeyValue(obj, "status", status)
+func WithStatus(t *testing.T, obj runtime.Object, status map[string]interface{}) runtime.Object {
+	res, err := WithKeyValue(obj, "status", status)
+	if err != nil {
+		t.Fatalf("failed to apply status %v to object %v: %v", status, obj, err)
+	}
+	return res
 }
 
 // WithKeyValue sets key in the provided object to value.
-func WithKeyValue(obj runtime.Object, key string, value map[string]interface{}) runtime.Object {
+func WithKeyValue(obj runtime.Object, key string, value map[string]interface{}) (runtime.Object, error) {
```

Comment:
This was fun: we would ignore two ways this method could fail return _the copy of the passed object_ instead. Which does not make the debugging of tests using this method easy.